gtk4/src/subclass/widget.rs

```diff
@@ -1245,6 +1245,43 @@ where
     }
 }
 
+impl<T> ToValue for TemplateChild<T>
+where
+    T: ToValue + ObjectType + FromGlibPtrNone<*mut <T as ObjectType>::GlibType>,
+{
+    #[inline]
+    fn to_value(&self) -> glib::Value {
+        T::to_value(&self.get())
+    }
+
+    #[inline]
+    fn value_type(&self) -> glib::Type {
+        T::static_type()
+    }
+}
+
+impl<T> glib::value::ValueType for TemplateChild<T>
+where
+    T: glib::value::ValueType + ObjectType + FromGlibPtrNone<*mut <T as ObjectType>::GlibType>,
+{
+    type Type = <T as glib::value::ValueType>::Type;
+}
+
+unsafe impl<'a, T> glib::value::FromValue<'a> for TemplateChild<T>
+where
+    T: ObjectType + FromGlibPtrNone<*mut <T as ObjectType>::GlibType>,
+{
+    type Checker = glib::value::GenericValueTypeChecker<T>;
+
+    #[inline]
+    unsafe fn from_value(value: &'a glib::Value) -> Self {
+        skip_assert_initialized!();
+        TemplateChild {
+            ptr: T::from_value(value).as_ptr(),
```

Comment:
Actually no, this looks wrong. The pointers becomes invalid after this call, you need `into_glib_ptr()` or so but then `TemplateChild` has no `Drop`...

Reply:
Right because Gtk is keeping the ref of the child and drops it itself. If we do implement Drop, we would have to some bool to identify whether we should unref the ptr or not. Sounds a bit hackish but no ideas how to do this properly :)

Reply:
Do we need the `FromValue` impl?

Reply:
Yes. Otherwise it can't really be a property

Reply:
Do we want writable properties to be possible for this? Seems strange to me. Readonly seems to make sense.

Reply:
Well, you want to have writable properties yes. All container widgets nowadays have a "child" property that is readwrite.

Reply:
Oh I see. Well, then you'll need a drop boolean :)

Reply:
Done that in the latest push